File: app/src/main/java/com/kaigarrott/stopwatchkt/data/TimeEntry.java
package com.kaigarrott.stopwatchkt.data;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.PrimaryKey;


@Entity(tableName = "time")
public class TimeEntry {

    @PrimaryKey(autoGenerate = true)
    private int id;
    private long timestamp;
    private long value;

    @Ignore
    public TimeEntry(long timestamp, long value) {
        this.timestamp = timestamp;
        this.value = value;
    }

    public TimeEntry(int id, long timestamp, long value) {
        this.id = id;
        this.timestamp = timestamp;
        this.value = value;
    }

    public int getId() {
        return id;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public long getValue() {
        return value;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }

    public void setValue(long value) {
        this.value = value;
    }
}

File: app/src/main/java/com/kaigarrott/stopwatchkt/data/TimeEntry.kt
package com.kaigarrott.stopwatchkt.data

import android.arch.persistence.room.Entity
import android.arch.persistence.room.Ignore
import android.arch.persistence.room.PrimaryKey


//@Entity(tableName = "time")
//class TimeEntry(@PrimaryKey(autoGenerate = true) val id: Int, val timestamp: Long, val value: Long)

@Entity(tableName = "time")
class TimeEntry {

    @PrimaryKey(autoGenerate = true)
    var id: Int = 0
    var timestamp: Long = 0
    var value: Long = 0

    @Ignore
    constructor(timestamp: Long, value: Long) {
        this.timestamp = timestamp
        this.value = value
    }

    constructor(id: Int, timestamp: Long, value: Long) {
        this.id = id
        this.timestamp = timestamp
        this.value = value
    }
}
